TM1py/Services/CellService.py

```diff
@@ -368,6 +370,57 @@ def write_dataframe(self, cube_name: str, data: 'pd.DataFrame', dimensions: Iter
                           use_changeset=use_changeset,
                           **kwargs)
 
+    @require_pandas
+    @manage_transaction_log
+    def write_dataframe_async(self, cube_name: str, data: 'pd.DataFrame',
+                              slice_size_of_dataframe: int, max_workers: int,
+                              increment: bool = False):
+        """ Write DataFrame into a cube using unbound TI processes in a multi-threading way. Requires admin permissions.
+        For a DataFrame with > 1,000,000 rows, this function will at least save half of runtime compared with `write_dataframe` function.
+        Column order must match dimensions in the target cube with an additional column for the values.
+        Column names are not relevant.
+        :param cube_name:
+        :param data: Pandas Data Frame
+        :param slice_size_of_dataframe: Number of rows for each DataFrame slice, e.g. 10000
+        :param max_workers: Max number of threads, e.g. 14
```

Comment:
We could add a note on the limit for max threads.

Reply:
Same as above.